smartsim/_core/_cli/build.py

```diff
@@ -248,14 +244,14 @@ def build_redis_ai(
 
         self.check_backends_install()
 
-        # Check for onnx and tf in user python environemnt and prompt user
-        # to download them if they are not installed. this should not break
-        # the build however, as we use onnx and tf directly from RAI instead
-        # of pip like we do PyTorch.
+        # Check for onnx and tf in user python environemnt or download
+        # them if requested. If they are not present and the user did not request
+        # they be downloaded, we still should not break the build, as we use
+        # onnx and tf directly from RAI instead of pip like we do PyTorch.
         if onnx:
-            self.check_onnx_install()
+            self.check_py_onnx_version(handle_conflict=self.modify_py_env)
```

Comment:
Minor quibble with the name: check_version doesn't sound like it should be doing an installation

Reply:
(applies to all backends)

Reply:
better named!!